routes/userRouter.js

```diff
@@ -1,107 +1,63 @@
 import express from 'express';
-import Joi from 'joi';
 import ejValidator from 'express-joi-validation';
+import userSchema from '../validation/schema/userSchema';
+import UserService from '../services/UserService';
+import userModel from '../models/UserModel';
 
 const userRouter = express.Router();
 const validator = ejValidator.createValidator();
 
-const bodySchema = Joi.object({
-  id: Joi.string().required(),
-  login: Joi.string().alphanum().min(3).max(30).required(),
-  password: Joi.string()
-    .regex(/^[a-zA-Z0-9]{3,30}$/)
-    .required(),
-  isDeleted: Joi.bool().required(),
-  age: Joi.number().integer().min(4).max(130).required(),
-});
-
-let users = [
-  {
-    id: '1',
-    login: 'user1',
-    password: 'password12',
-    age: 26,
-    isDeleted: false,
-  },
-];
+const userService = new UserService(userModel);
 
-userRouter.get('/users/:id', (req, res) => {
-  const { id } = req.params;
-  console.log(users);
-  const user = users.find((user) => user.id === id);
-  if (user) {
+userRouter.get('/user/:id', async (req, res) => {
+  try {
+    const { id } = req.params;
+    const user = await userService.getUserById(id);
     res.send(user);
-  } else {
-    res.send('User not found');
+  } catch (error) {
+    res.status(500).send(error.message);
```

Comment:
A good way to throw errors is not to hardcode error status codes.
Rather make it generic:
`res.status(err.response.status).send(err.response.statusText);`


eg:
Scenario 1: id passed in req.params is blank, we can throw 400 for that, like:
```
throw new BadRequestError({
        response: { status: 400, statusText: 'Invalid user passed' },
});
```
So, for that you can create common error class and throw errors like this:
`throw new BadRequestError('Invalid user Id passed ' + error);`
Your code will flow in catch block and appropriate error response with status will throw.


Scenario 2:
Your DB is not responding in Models so you can throw timeout error with 500 status code and your route class will catch it and will show the proper error response with status code.

Please consider checking all your catch blocks for 500 error code as for the wrong Id passed it will throw 500 whereas 5xx series are generally related to infra failure errors and 4xx are related to not found ones. 
I guess you have a further session on Error Handling but it is a good attempt.